src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import duke.command.CommandNotFoundException;
+import duke.parser.IncorrectArgumentsException;
+import duke.parser.IncorrectFileFormatException;
+import duke.parser.IncorrectNumberOfArgumentsException;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+import duke.parser.Parser;
+import duke.command.Command;
+
+/**
+ * Represents a Duke - interactive bot.
+ * Contains functions to operate the bot.
+ */
+public class Duke {
+	private Storage storage;
+	private TaskList tasks;
+	private Ui ui;
+	
+	/**
+	 * Constructor to initialize Duke for startup.
+	 * Set up storage, parser and ui.
+	 */
+	public Duke() {
+		Parser.initialize();
+		storage = new Storage();
+		ui = new Ui();
+	}
+	
+	/**
+	 * Performs Duke start up operation to generate task list.
+	 */
+	public void performDukeStartup() throws IncorrectFileFormatException, FileNotFoundException,
+											IncorrectArgumentsException {
+		tasks = new TaskList();
+		try {
+			tasks = new TaskList(storage.load(ui));
+		} catch (IncorrectFileFormatException j) {
```

Comment:
perhaps the exceptions pertaining to the file to be loaded can be resolved in Storage class instead?